analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseDateTimeInsteadOfDateTimeOffset.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Globalization;
+
+public class Program
+{
+    void Constructors()
+    {
+        _ = new DateTime(1);                                                                 // Noncompliant {{Prefer using "DateTimeOffset" struct instead of "DateTime"}}
+//          ^^^^^^^^^^^^^^^
+        _ = new DateTime(1, 1, 1);                                                           // Noncompliant
+        _ = new DateTime(1, 1, 1, new GregorianCalendar());                                  // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1);                                                  // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, new GregorianCalendar());                         // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, DateTimeKind.Utc);                                // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, 1);                                               // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, 1, new GregorianCalendar());                      // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, 1, DateTimeKind.Utc);                             // Noncompliant
+        _ = new DateTime(1, 1, 1, 1, 1, 1, 1, new GregorianCalendar(), DateTimeKind.Utc);    // Noncompliant
+    }
+
+    void Fields()
+    {
+        _ = DateTime.MaxValue;  // Noncompliant
+//          ^^^^^^^^^^^^^^^^^
```

Comment:
The message is about using one type instead of another. So I'd only highlight the type name / declaration, rather than the whole member access expression (it's usage).